server/src/action/user/create.js

```diff
@@ -0,0 +1,44 @@
+const AbstractAction = require('../abstract')
+const { body } = require('express-validator')
+const capitalize = require('../sanitizer/capitalize')
+const uppercase = require('../sanitizer/uppercase')
+
+class CreateUser extends AbstractAction {
+  get validate () {
+    return [
+      body('invitation')
+        .exists({ checkNull: true }).withMessage('required field')
+        .isString().withMessage('must be a string')
+        .trim(),
+      body('firstname')
+        .exists({ checkNull: true }).withMessage('required field')
+        .isString().withMessage('must be a string')
+        .trim()
+        .isLength({ min: 1, max: 255 }).withMessage('length must be between 1 and 255')
+        .customSanitizer(capitalize),
+      body('lastname')
+        .exists({ checkNull: true }).withMessage('required field')
+        .isString().withMessage('must be a string')
+        .trim()
+        .isLength({ min: 1 }).withMessage('length must be greater than 1')
+        .customSanitizer(uppercase),
+      body('password')
+        .exists({ checkNull: true }).withMessage('required field')
```

Comment:
The `bcrypt` implementation takes input of 72 characters maximum. Any longer password will be truncated.

I think we should limit the password's length to 72. I'm not a fan of that (i like password to have an infinite length), but unless you're ready to switch to something that also accepts infinite input i really  do not wanna give a false sensation of security to the user.

Reply:
72 is more than enough, we don't need infinite length password we are not a nuclear plant...
To ensure the password is strong enough, I think I'll check that it has at least 8 characters and it contains a number, a lowercase and an uppercase

Reply:
I'm not saying it's not enough, i'm suggesting you should add a hard limit that prevents user creation if the password is longer.

Reply:
yes it will be set

Reply:
(server side)